Sources/SmokeDynamoDB/AWSDynamoDBClientConfiguration.swift

```diff
@@ -0,0 +1,195 @@
+// Copyright 2018-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// AWSDynamoDBClientConfiguration.swift
+// DynamoDBClient
+//
+
+import DynamoDBModel
+import DynamoDBClient
+import SmokeAWSCore
+import SmokeHTTPClient
+import NIO
+import SmokeAWSHttp
+import AsyncHTTPClient
+import Logging
+
+internal extension SmokeHTTPClient.HTTPClientError {
+    func isRetriable() -> Bool {
+        if let typedError = self.cause as? DynamoDBError, let isRetriable = typedError.isRetriable() {
+            return isRetriable
+        } else {
+            return self.isRetriableAccordingToCategory
+        }
+    }
+}
+
+public typealias AWSDynamoDBClientConfiguration =
+    AWSGenericDynamoDBClientConfiguration<StandardHTTPClientCoreInvocationReporting<AWSClientInvocationTraceContext>>
+
+/**
+ Configuration for an AWS DynamoDB client.
+ */
+public struct AWSGenericDynamoDBClientConfiguration<InvocationReportingType: HTTPClientCoreInvocationReporting> {
+    public let endpointHostName: String
+    public let endpointPort: Int
+    public let contentType: String
+    public let timeoutConfiguration: HTTPClient.Configuration.Timeout
+    public let connectionPoolConfiguration: HTTPClient.Configuration.ConnectionPool?
+    public let awsRegion: AWSRegion
+    public let retryConfiguration: HTTPClientRetryConfiguration
+    public let eventLoopGroup: EventLoopGroup
+    public let reportingConfiguration: SmokeAWSClientReportingConfiguration<DynamoDBModelOperations>
+    
+    internal let clientDelegate: JSONAWSHttpClientDelegate<DynamoDBError>
+    internal let reportingProvider: (Logger, String, EventLoop?) -> InvocationReportingType
+    internal let credentialsProvider: CredentialsProvider
+    
+    public init<NewTraceContextType: InvocationTraceContext>(
```

Comment:
What does "new" in `NewTraceContextType` refer to?

Reply:
It was a copy/paste error. Fixed.